api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,130 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := 0
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		if err, ok := err.(*errors.ServerError); ok && err.Code == errors.ServiceDoesNotExist {
+			return nil
+		}
+
+		return err
+	}
+
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	service, err := s.readService(clusterName, fqServiceID)
+	if err != nil {
+		return err
+	}
+
+	taskARNs, err := s.getServiceActiveTasks(clusterName, service.Deployments)
+	if err != nil {
+		return err
+	}
+
+	if err := s.stopServiceTasks(clusterName, taskARNs); err != nil {
+		return err
+	}
+
+	if err := s.scaleService(clusterName, fqServiceID, desiredCount); err != nil {
+		return err
+	}
+
+	if err := s.deleteService(clusterName, fqServiceID); err != nil {
+		return err
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) getServiceActiveTasks(clusterName string, deployments []*ecs.Deployment) ([]string, error) {
+	taskARNs := []string{}
+
+	for _, deployment := range deployments {
+		clusterTaskARNsStopped, err := listClusterTaskARNs(s.AWS.ECS, clusterName, aws.StringValue(deployment.Id), ecs.DesiredStatusStopped)
```

Comment:
Need to remove fetching the stopped tasks

Reply:
Okay, this was carried over from the legacy, so I wasn't sure why it was there before.